### DISCUSSION TITLE
automatically warm-start the second solve call

### DISCUSSION THREAD

**Discussion by BrannonKing** [google/or-tools#4433] [CP-SAT questions]
It seems to me that CpSolver could automatically warm-start itself when called a second time with the same model. It appears to not do this at present. I was hoping to be able to just modify the objective on the model and pass it in a second time, having it automatically utilize its previous findings. (CpSolver struggles to find and initial feasible solution on my model, but it seems to do better after it has one.) What is the recommended way to warm-start it in this scenario? Just use the hints?

**Comment by lperron** [comment]
yes. Solve is stateless.
